File: SOURCE/SmartParkingAPI/Models/Entities/ParkingSlot.cs
using SmartParkingAPI.Models.Contracts;

namespace SmartParkingAPI.Models.Entities
{
    public class ParkingSlot : IEntity
    {
        public int Id { get; set; }
        public int ParkingLotId { get; set; }
        public string? SlotCode { get; set; }
        public bool IsOccupied { get; set; }

        public ParkingLot? ParkingLot { get; set; }
    }
}

File: SOURCE/smart_parking_app/lib/models/parking_slot.dart
class ParkingSlot {
  final int id;
  final String slotCode;
  final bool isOccupied;
  final int parkingLotId;

  ParkingSlot({
    required this.id,
    required this.slotCode,
    required this.isOccupied,
    required this.parkingLotId,
  });

  factory ParkingSlot.fromJson(Map<String, dynamic> json) {
    return ParkingSlot(
      id: json['id'],
      slotCode: json['slotCode'],
      isOccupied: json['isOccupied'],
      parkingLotId: json['parkingLotId'],
    );
  }
}
